src/components/engagement/events/engagement-created.event.ts

```diff
@@ -0,0 +1,11 @@
+import { ISession } from '../../../common';
+import { CreateCeremony } from '../../ceremony/dto';
+import { LanguageEngagement } from '../dto';
+
+export class EngagementCreatedEvent {
+  constructor(
+    readonly languageEngagement: LanguageEngagement,
```

Comment:
Considering the updated and deleted events just use the `Engagement` type we should follow suit here.
```suggestion
    readonly engagement: Engagement,
```

Reply:
I have refactored. Now the code is more clear.